src/components/SidebarState.vue

```diff
@@ -43,19 +43,37 @@ export default Vue.extend({
     },
     methods: {
         section() {
-            if (this.isClosed) {
+            if (!this.sidebarOpen) {
                 return '';
             }
 
             let section = this.sidebarSection;
-            let isChannel = this.$state.getActiveBuffer().isChannel();
+            let buffer = this.$state.getActiveBuffer();
 
-            if (section === 'settings' && isChannel) {
-                return 'settings';
-            } else if (section === 'user' && this.sidebarUser && isChannel) {
-                return 'user';
-            } else if (section === 'nicklist' && isChannel) {
+            if (buffer.isQuery()) {
+                let user = this.$state.getUser(buffer.getNetwork().id, buffer.name);
+                if (user) {
+                    this.sidebarUser = user;
+                    return 'user';
+                }
+                return '';
+            }
+
+            // The following code is for channels only
+            if (!buffer.isChannel()) {
+                return '';
+            }
+
+            if (section === 'user' && this.sidebarUser) {
```

Comment:
This is checking for a user despite the user section handling this a few lines up. Should this be needed here?

Reply:
this extra check is for when switching from a pm back to a channel where that user is not present, it will instead show the nicklist, ive improved this to also change the state so it doesnt jump again when switching to a channel with the last user you opened.

also improved the comment here explaining why